polymorphia-backend/src/main/java/com/agh/polymorphia_backend/repository/gradable_event/GradableEventRepository.java

```diff
@@ -6,42 +6,124 @@
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
-import java.math.BigDecimal;
+import java.util.List;
 import java.util.Optional;
+import com.agh.polymorphia_backend.repository.gradable_event.projections.InstructorGradableEventProjection;
+import com.agh.polymorphia_backend.repository.gradable_event.projections.StudentGradableEventProjection;
+
 
 @Repository
 public interface GradableEventRepository extends JpaRepository<GradableEvent, Long> {
+
     @Query("SELECT ge.eventSection.course.id FROM GradableEvent ge WHERE ge.id = :id")
     Optional<Long> getCourseIdByGradableEventId(@Param("id") Long id);
 
     @Query("""
-                SELECT COUNT(cr) > 0
-                FROM CriterionReward cr
-                WHERE cr.criterion.gradableEvent = :gradableEvent
-            """)
-    boolean rewardExists(GradableEvent gradableEvent);
+    SELECT ge.id as id,
+           ge.name as name,
+           ge.orderIndex as orderIndex,
+           ge.topic as topic,
+           ge.roadMapOrderIndex as roadMapOrderIndex,
+           ge.isHidden as isHidden,
+           ge.isLocked as isLocked,
+           CASE
+               WHEN COUNT(DISTINCT g.id) > 0 
+               THEN TO_CHAR(SUM(DISTINCT cg.xp), 'FM9999990.0')
+               ELSE NULL
+           END as gainedXp,
+           CASE WHEN COUNT(DISTINCT cr.criterion.id) > 0 THEN true ELSE false END as hasPossibleReward,
+           CASE WHEN COUNT(DISTINCT g.id) > 0 THEN true ELSE false END as isGraded,
+           CASE WHEN COUNT(DISTINCT g.id) > 0 AND COUNT(DISTINCT ar.id) > 0 
+                THEN true 
+                ELSE false 
+           END as isRewardAssigned
```

Comment:
```suggestion
           COUNT(DISTINCT g.id) > 0 as isGraded,
           COUNT(DISTINCT g.id) > 0 AND COUNT(DISTINCT ar.id) > 0 as isRewardAssigned
```
this should also work